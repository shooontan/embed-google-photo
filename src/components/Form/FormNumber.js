// @flow
import * as React from 'react';

type Props = {
  label?: string,
};

const FormNumber = (props: Props) => {
  const { label } = props;
  return (
    <p>
      <input type="number" {...props} />
      {label}
    </p>
  );
};

FormNumber.defaultProps = {
  label: '',
};

export default FormNumber;
