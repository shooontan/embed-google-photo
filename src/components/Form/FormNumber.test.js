// @flow
import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import FormNumber from './FormNumber';

Enzyme.configure({ adapter: new Adapter() });

test('FormNumber Component', () => {
  let testRenderer = TestRenderer.create(<FormNumber />);
  expect(testRenderer).toMatchSnapshot();

  testRenderer = TestRenderer.create(<FormNumber label="form number" />);
  expect(testRenderer).toMatchSnapshot();

  const mock = jest.fn();
  testRenderer = shallow(<FormNumber onChange={mock} />);
  testRenderer.find('input').simulate('change');
  expect(mock).toHaveBeenCalled();
});
