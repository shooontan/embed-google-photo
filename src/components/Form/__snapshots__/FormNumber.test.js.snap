// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormNumber Component 1`] = `
<p>
  <input
    label=""
    type="number"
  />
  
</p>
`;

exports[`FormNumber Component 2`] = `
<p>
  <input
    label="form number"
    type="number"
  />
  form number
</p>
`;
